### PR TITLE
add crkbd/keymaps/bermeo

### PR DESCRIPTION
**Pull request by GMBermeo**
This layout is mainly focused on 3 groups:
* Romance (neo-Latin) speakers/writers. Like Portuguese, Spanish, Italian, and French.
* UX/UI/Front-End Designers: heavy usage of mouse and left-hand keyboard modifiers in Adobe, Spline, Figma, and other visual software.
* Someone that wants to configure per-key LED colors, even for layers and modifiers.

## Description

Heavy usage of RGB Matrix LED colors, and OLED programming, with fully animated Luna, far beyond the original.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by GMBermeo** [comment]
Compiled and flashed:

```
qmk_bermeo % make crkbd:bermeo:flash
QMK Firmware 0.16.9
Making crkbd/rev1 with keymap bermeo and target flash

avr-gcc (Homebrew AVR GCC 8.4.0_2) 8.4.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Generating: .build/obj_crkbd_rev1/src/info_config.h                                                 [OK]
Generating: .build/obj_crkbd_rev1/src/default_keyboard.h                                            [OK]
Generating: .build/obj_crkbd_rev1/src/layouts.h                                                     [OK]
Compiling: keyboards/crkbd/crkbd.c                                                                  [OK]
Compiling: keyboards/crkbd/rev1/rev1.c                                                              [OK]
Compiling: keyboards/crkbd/keymaps/bermeo/keymap.c                                                  [OK]
Compiling: quantum/quantum.c                                                                        [OK]
Compiling: quantum/send_string.c                                                                    [OK]
Compiling: quantum/bitwise.c                                                                        [OK]
Compiling: quantum/led.c                                                                            [OK]
Compiling: quantum/action.c                                                                         [OK]
Compiling: quantum/action_layer.c                                                                   [OK]
Compiling: quantum/action_tapping.c                                                                 [OK]
Compiling: quantum/action_util.c                                                                    [OK]
Compiling: quantum/eeconfig.c                                                                       [OK]
Compiling: quantum/keyboard.c                                                                       [OK]
Compiling: quantum/keymap_common.c                                                                  [OK]
Compiling: quantum/keycode_config.c                                                                 [OK]
Compiling: quantum/sync_timer.c                                                                     [OK]
Compiling: quantum/logging/debug.c                                                                  [OK]
Compiling: quantum/logging/sendchar.c                                                               [OK]
Compiling: quantum/bootmagic/magic.c                                                                [OK]
Compiling: quantum/matrix_common.c                                                                  [OK]
Compiling: quantum/matrix.c                                                                         [OK]
Compiling: quantum/debounce/sym_defer_g.c                                                           [OK]
Compiling: quantum/split_common/split_util.c                                                        [OK]
Compiling: quantum/split_common/transport.c                                                         [OK]
Compiling: quantum/split_common/transactions.c                                                      [OK]
Compiling: quantum/main.c                                                                           [OK]
Compiling: quantum/color.c                                                                          [OK]
Compiling: quantum/rgb_matrix/rgb_matrix.c                                                          [OK]
Compiling: quantum/rgb_matrix/rgb_matrix_drivers.c                                                  [OK]
Compiling: lib/lib8tion/lib8tion.c                                                                  [OK]
Compiling: quantum/process_keycode/process_rgb.c                                                    [OK]
Compiling: platforms/avr/sleep_led.c                                                                [OK]
Compiling: platforms/avr/drivers/ws2812.c                                                           [OK]
Compiling: quantum/led_tables.c                                                                     [OK]
Compiling: quantum/crc.c                                                                            [OK]
Compiling: drivers/oled/ssd1306_sh1106.c                                                            [OK]
Compiling: quantum/process_keycode/process_magic.c                                                  [OK]
Compiling: quantum/process_keycode/process_grave_esc.c                                              [OK]
Compiling: quantum/process_keycode/process_space_cadet.c                                            [OK]
Compiling: quantum/wpm.c                                                                            [OK]
Compiling: platforms/avr/drivers/i2c_master.c                                                       [OK]
Archiving: .build/obj_crkbd_rev1_bermeo/i2c_master.o                                                [OK]
Compiling: platforms/avr/drivers/i2c_slave.c                                                        [OK]
Archiving: .build/obj_crkbd_rev1_bermeo/i2c_slave.o                                                 [OK]
Compiling: platforms/avr/drivers/serial.c                                                           [OK]
Archiving: .build/obj_crkbd_rev1_bermeo/serial.o                                                    [OK]
Assembling: platforms/avr/xprintf.S                                                                 [OK]
Compiling: platforms/avr/printf.c                                                                   [OK]
Compiling: tmk_core/protocol/host.c                                                                 [OK]
Compiling: tmk_core/protocol/report.c                                                               [OK]
Compiling: tmk_core/protocol/usb_device_state.c                                                     [OK]
Compiling: tmk_core/protocol/usb_util.c                                                             [OK]
Compiling: platforms/suspend.c                                                                      [OK]
Compiling: platforms/avr/platform.c                                                                 [OK]
Compiling: platforms/avr/suspend.c                                                                  [OK]
Compiling: platforms/avr/timer.c                                                                    [OK]
Compiling: platforms/avr/bootloaders/caterina.c                                                     [OK]
Compiling: tmk_core/protocol/lufa/lufa.c                                                            [OK]
Compiling: tmk_core/protocol/usb_descriptor.c                                                       [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Class/Common/HIDParser.c                                       [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/Device_AVR8.c                                        [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/EndpointStream_AVR8.c                                [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/Endpoint_AVR8.c                                      [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/Host_AVR8.c                                          [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/PipeStream_AVR8.c                                    [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/Pipe_AVR8.c                                          [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/USBController_AVR8.c                                 [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/USBInterrupt_AVR8.c                                  [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/ConfigDescriptors.c                                       [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/DeviceStandardReq.c                                       [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/Events.c                                                  [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/HostStandardReq.c                                         [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/USBTask.c                                                 [OK]
Compiling: tmk_core/protocol/lufa/usb_util.c                                                        [OK]
Linking: .build/crkbd_rev1_bermeo.elf                                                               [OK]
Creating load file for flashing: .build/crkbd_rev1_bermeo.hex                                       [OK]
Copying crkbd_rev1_bermeo.hex to qmk_firmware folder                                                [OK]
Checking file size of crkbd_rev1_bermeo.hex                                                         [WARNINGS]

 * The firmware size is approaching the maximum - 28658/28672 (99%, 14 bytes free)
Waiting for USB serial port - reset your controller now (Ctrl+C to cancel).........................
Device /dev/tty.usbmodem14301 has appeared; assuming it is the controller.
Waiting for /dev/tty.usbmodem14301 to become writable.

Connecting to programmer: .
Found programmer: Id = "CATERIN"; type = S
    Software Version = 1.0; No Hardware Version given.
Programmer supports auto addr increment.
Programmer supports buffered memory access with buffersize=128 bytes.

Programmer supports the following devices:
    Device code: 0x44

avrdude: AVR device initialized and ready to accept instructions

Reading | ################################################## | 100% 0.00s

avrdude: Device signature = 0x1e9587 (probably m32u4)
avrdude: NOTE: "flash" memory has been specified, an erase cycle will be performed
         To disable this feature, specify the -D option.
avrdude: erasing chip
avrdude: reading input file ".build/crkbd_rev1_bermeo.hex"
avrdude: input file .build/crkbd_rev1_bermeo.hex auto detected as Intel Hex
avrdude: writing flash (28658 bytes):

Writing | ################################################## | 100% 2.16s

avrdude: 28658 bytes of flash written
avrdude: verifying flash memory against .build/crkbd_rev1_bermeo.hex:
avrdude: input file .build/crkbd_rev1_bermeo.hex auto detected as Intel Hex

Reading | ################################################## | 100% 0.23s

avrdude: 28658 bytes of flash verified

avrdude done.  Thank you.

```

**Review by GMBermeo — COMMENTED** [review]
Compiling fine.
```
$/qmk_firmware % make crkbd:bermeo
QMK Firmware 0.16.9
Making crkbd/rev1 with keymap bermeo

avr-gcc (Homebrew AVR GCC 8.4.0_2) 8.4.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Size before:
   text    data     bss     dec     hex filename
      0   28658       0   28658    6ff2 crkbd_rev1_bermeo.hex

Compiling: keyboards/crkbd/crkbd.c                                                                  [OK]
Compiling: keyboards/crkbd/rev1/rev1.c                                                              [OK]
Compiling: keyboards/crkbd/keymaps/bermeo/keymap.c                                                  [OK]
Compiling: quantum/quantum.c                                                                        [OK]
Compiling: quantum/send_string.c                                                                    [OK]
Compiling: quantum/bitwise.c                                                                        [OK]
Compiling: quantum/led.c                                                                            [OK]
Compiling: quantum/action.c                                                                         [OK]
Compiling: quantum/action_layer.c                                                                   [OK]
Compiling: quantum/action_tapping.c                                                                 [OK]
Compiling: quantum/action_util.c                                                                    [OK]
Compiling: quantum/eeconfig.c                                                                       [OK]
Compiling: quantum/keyboard.c                                                                       [OK]
Compiling: quantum/keymap_common.c                                                                  [OK]
Compiling: quantum/keycode_config.c                                                                 [OK]
Compiling: quantum/sync_timer.c                                                                     [OK]
Compiling: quantum/logging/debug.c                                                                  [OK]
Compiling: quantum/logging/sendchar.c                                                               [OK]
Compiling: quantum/bootmagic/magic.c                                                                [OK]
Compiling: quantum/matrix_common.c                                                                  [OK]
Compiling: quantum/matrix.c                                                                         [OK]
Compiling: quantum/debounce/sym_defer_g.c                                                           [OK]
Compiling: quantum/split_common/split_util.c                                                        [OK]
Compiling: quantum/split_common/transport.c                                                         [OK]
Compiling: quantum/split_common/transactions.c                                                      [OK]
Compiling: quantum/main.c                                                                           [OK]
Compiling: quantum/color.c                                                                          [OK]
Compiling: quantum/rgb_matrix/rgb_matrix.c                                                          [OK]
Compiling: quantum/rgb_matrix/rgb_matrix_drivers.c                                                  [OK]
Compiling: lib/lib8tion/lib8tion.c                                                                  [OK]
Compiling: quantum/process_keycode/process_rgb.c                                                    [OK]
Compiling: platforms/avr/sleep_led.c                                                                [OK]
Compiling: platforms/avr/drivers/ws2812.c                                                           [OK]
Compiling: quantum/led_tables.c                                                                     [OK]
Compiling: quantum/crc.c                                                                            [OK]
Compiling: drivers/oled/ssd1306_sh1106.c                                                            [OK]
Compiling: quantum/process_keycode/process_magic.c                                                  [OK]
Compiling: quantum/process_keycode/process_grave_esc.c                                              [OK]
Compiling: quantum/process_keycode/process_space_cadet.c                                            [OK]
Compiling: quantum/wpm.c                                                                            [OK]
Compiling: platforms/avr/drivers/i2c_master.c                                                       [OK]
Archiving: .build/obj_crkbd_rev1_bermeo/i2c_master.o                                                [OK]
Compiling: platforms/avr/drivers/i2c_slave.c                                                        [OK]
Archiving: .build/obj_crkbd_rev1_bermeo/i2c_slave.o                                                 [OK]
Compiling: platforms/avr/drivers/serial.c                                                           [OK]
Archiving: .build/obj_crkbd_rev1_bermeo/serial.o                                                    [OK]
Compiling: platforms/avr/printf.c                                                                   [OK]
Compiling: tmk_core/protocol/host.c                                                                 [OK]
Compiling: tmk_core/protocol/report.c                                                               [OK]
Compiling: tmk_core/protocol/usb_device_state.c                                                     [OK]
Compiling: tmk_core/protocol/usb_util.c                                                             [OK]
Compiling: platforms/suspend.c                                                                      [OK]
Compiling: platforms/avr/platform.c                                                                 [OK]
Compiling: platforms/avr/suspend.c                                                                  [OK]
Compiling: platforms/avr/timer.c                                                                    [OK]
Compiling: platforms/avr/bootloaders/caterina.c                                                     [OK]
Compiling: tmk_core/protocol/lufa/lufa.c                                                            [OK]
Compiling: tmk_core/protocol/usb_descriptor.c                                                       [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Class/Common/HIDParser.c                                       [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/Device_AVR8.c                                        [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/EndpointStream_AVR8.c                                [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/Endpoint_AVR8.c                                      [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/Host_AVR8.c                                          [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/PipeStream_AVR8.c                                    [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/Pipe_AVR8.c                                          [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/USBController_AVR8.c                                 [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/USBInterrupt_AVR8.c                                  [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/ConfigDescriptors.c                                       [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/DeviceStandardReq.c                                       [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/Events.c                                                  [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/HostStandardReq.c                                         [OK]
Compiling: lib/lufa/LUFA/Drivers/USB/Core/USBTask.c                                                 [OK]
Compiling: tmk_core/protocol/lufa/usb_util.c                                                        [OK]
Linking: .build/crkbd_rev1_bermeo.elf                                                               [OK]
Creating load file for flashing: .build/crkbd_rev1_bermeo.hex                                       [OK]
Copying crkbd_rev1_bermeo.hex to qmk_firmware folder                                                [OK]
Checking file size of crkbd_rev1_bermeo.hex                                                         [WARNINGS]

 * The firmware size is approaching the maximum - 28658/28672 (99%, 14 bytes free)
```

File: keyboards/crkbd/keymaps/bermeo/README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# Bermeo Layout for Corne
```
Comment (drashna) [inline keyboards/crkbd/keymaps/bermeo/README.md:1]: Please rename this file to be all lower case.

File: keyboards/crkbd/keymaps/bermeo/keymap.c [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
+
+#ifdef RGBLIGHT_ENABLE
+// Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+#endif
```
Comment (drashna) [inline keyboards/crkbd/keymaps/bermeo/keymap.c:11]:
```suggestion
```

File: keyboards/crkbd/keymaps/bermeo/keymap.c [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+enum layers { _QWERTY, _LOWER, _RAISE, _NUMP };
```
Comment (drashna) [inline keyboards/crkbd/keymaps/bermeo/keymap.c:17]:
```suggestion
enum layers { 
    _QWERTY, 
    _LOWER, 
    _RAISE, 
    _NUMP,
};
```

File: keyboards/crkbd/keymaps/bermeo/keymap.c [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
+    )};
+
+int RGB_current_mode;
+
+void matrix_init_user(void) {
+#ifdef RGBLIGHT_ENABLE
+    RGB_current_mode = rgblight_config.mode;
+#endif
+}
+
```
Comment (drashna) [inline keyboards/crkbd/keymaps/bermeo/keymap.c:90]:
```suggestion
```

File: keyboards/crkbd/keymaps/bermeo/keymap.c [hunk 5]
```diff
@@ -0,0 +1,603 @@
...
+//Per key lights
+void rgb_matrix_indicators_user(void) {
+#ifdef RGB_MATRIX_ENABLE
+    switch (biton32(layer_state)) {
```
Comment (drashna) [inline keyboards/crkbd/keymaps/bermeo/keymap.c:94]:
```suggestion
    switch (get_highest_layer(layer_state)) {
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by GMBermeo — COMMENTED** [review]
All changes requested by @drashna, tested and working with `make crkbd:bermeo`.

**Review by GMBermeo — COMMENTED** [review]
Compiling with make crkbd:bermeo
```
% make crkbd:bermeo
QMK Firmware 0.16.9
Making crkbd/rev1 with keymap bermeo

avr-gcc (Homebrew AVR GCC 8.4.0_2) 8.4.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Size before:
   text	   data	    bss	    dec	    hex	filename
      0	  28658	      0	  28658	   6ff2	crkbd_rev1_bermeo.hex

Copying crkbd_rev1_bermeo.hex to qmk_firmware folder                                                [OK]
Checking file size of crkbd_rev1_bermeo.hex                                                         [WARNINGS]

 * The firmware size is approaching the maximum - 28658/28672 (99%, 14 bytes free)
```

File: keyboards/crkbd/keymaps/bermeo/config.h [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+#define MASTER_LEFT
+// #define MASTER_RIGHT
+// #define EE_HANDS
+
+// #define SSD1306OLED
+#undef USE_I2C
+#undef SSD1306OLED
+
+#define USE_SERIAL_PD2
+
```
Comment (drashna) [inline keyboards/crkbd/keymaps/bermeo/config.h:36]:
```suggestion
```

File: keyboards/crkbd/keymaps/bermeo/config.h [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+// #define TAPPING_FORCE_HOLD
+#define TAPPING_TERM 150
+// #define RETRO_TAPPING
+// #define IGNORE_MOD_TAP_INTERRUPT
+// #define PERMISSIVE_HOLD
+
+#ifdef RGBLIGHT_ENABLE
+#    undef RGBLED_NUM
+#    define RGBLED_NUM 27
+#    define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/crkbd/keymaps/bermeo/config.h:46]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

Note: you don't need (and probably don't want) all of these modes enabled, and that will save on some space.

**Review by drashna — CHANGES_REQUESTED** [review]
You might also want to check out this, if you haven't already: https://docs.qmk.fm/#/squeezing_avr

**Review by GMBermeo — COMMENTED** [review]
Worked fine with changes of matrix rgb definitions.

**Review by drashna — APPROVED** [review]

**Comment by GMBermeo** [comment]
Ready to merge.

**Review by tzarc — APPROVED** [review]

**Pull request merged**